### PR TITLE
added comments trap.[ch]

### PR DESCRIPTION
**Pull request by kim6515516**
Added comments as doxygen style into trcp.[ch]

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

File: hypervisor/trap.c [hunk 1]
```diff
@@ -7,21 +7,43 @@
 #include <vdev.h>
 #define DEBUG
 #include <log/print.h>
-
+/**\defgroup Defines variables.
```
Comment (Wonseok) [inline hypervisor/trap.c:10]: Is variables mean ARM registers?
Please use domain-specific name.

File: hypervisor/trap.c [hunk 2]
```diff
@@ -7,21 +7,43 @@
...
+ * <pre> ARM registers.
+ * ARM registers include 13 general purpose register r0-r12, 1 Stack Pointer,
```
Comment (Wonseok) [inline hypervisor/trap.c:12]: s/13 general purpose register/13 general purpose registers

File: hypervisor/trap.c [hunk 3]
```diff
@@ -7,21 +7,43 @@
...
+ * 1 Link Register (LR), 1 Program Counter (PC).
+ * </pre>
+ */
+/**
+ * @brief Handles data abort exception taken from a mode other in Hyp mode.
+ * This fuction uses current ARM registers to dump and save as parameter.
+ * @param regs Current ARM registers.
```
Comment (Wonseok) [inline hypervisor/trap.c:19]: The ARM registers is used by current virtual machine, isn't it?
We have to think it more.

File: hypervisor/trap.c [hunk 4]
```diff
@@ -7,21 +7,43 @@
...
 hvmm_status_t _hyp_dabort(struct arch_regs *regs)
 {
     guest_dump_regs(regs);
     hyp_abort_infinite();
     return HVMM_STATUS_UNKNOWN_ERROR;
 }
-
+/**
+ * @brief Handles IRQ exception when interrupt is occured by device.
+ * This fucntion called gic interrupt and switched context. also this fuction
```
Comment (Wonseok) [inline hypervisor/trap.c:30]: s/also/Also

File: hypervisor/trap.c [hunk 5]
```diff
@@ -7,21 +7,43 @@
...
+ * uses current ARM registers to dump and save as parameter.
```
Comment (Wonseok) [inline hypervisor/trap.c:31]: Where is the function for dump and save as parameter in this code?

File: hypervisor/trap.c [hunk 6]
```diff
@@ -7,21 +7,43 @@
...
+ * <pre> ARM registers.
+ * ARM registers include 13 general purpose register r0-r12, 1 Stack Pointer,
+ * 1 Link Register (LR), 1 Program Counter (PC).
+ * </pre>
+ */
+/**
+ * @brief Handles data abort exception taken from a mode other in Hyp mode.
+ * This fuction uses current ARM registers to dump and save as parameter.
+ * @param regs Current ARM registers.
+ * @return It doesn't reach last step due to infinte loop.
```
Comment (Wonseok) [inline hypervisor/trap.c:20]: Returns HVMM_STATUS_UNKNOWN_ERROR only. We don't expect this exception now.

File: hypervisor/trap.c [hunk 7]
```diff
@@ -7,21 +7,43 @@
...
     hyp_abort_infinite();
     return HVMM_STATUS_UNKNOWN_ERROR;
 }
-
+/**
+ * @brief Handles IRQ exception when interrupt is occured by device.
+ * This fucntion called gic interrupt and switched context. also this fuction
+ * uses current ARM registers to dump and save as parameter.
+ * @param regs Current ARM registers.
+ * @return It always returns HVMM_STATUS_SUCCESS.
```
Comment (Wonseok) [inline hypervisor/trap.c:33]: Returns HVMM_STATUS_SUCCESS only.

File: hypervisor/trap.c [hunk 8]
```diff
@@ -7,21 +7,43 @@
...
+ */
 hvmm_status_t _hyp_irq(struct arch_regs *regs)
 {
     gic_interrupt(0, regs);
     guest_perform_switch(regs);
     return HVMM_STATUS_SUCCESS;
 }
-
+/**
+ * @brief Handles unhandled exception. unhanled exception has three category in
```
Comment (Wonseok) [inline hypervisor/trap.c:42]: Handles undefined, hypervisor, prefetch abort as unhandled exception. We don't support these exceptions now.

File: hypervisor/trap.c [hunk 9]
```diff
@@ -29,12 +51,18 @@ hvmm_status_t _hyp_unhandled(struct arch_regs *regs)
 
     return HVMM_STATUS_UNKNOWN_ERROR;
 }
-
+/**
+ * @brief Indirecting _hyp_hvc_service function in file.
```
Comment (Wonseok) [inline hypervisor/trap.c:55]: You must not start a comment with gerund.

File: hypervisor/trap.c (hunk 8 above)
Comment (Wonseok) [inline hypervisor/trap.c:42]: HVC is not hypervisor abort?

File: hypervisor/trap.h [hunk 10]
```diff
@@ -92,8 +92,19 @@
 #define HPFAR_FIPA_SHIFT                    4
 #define HPFAR_FIPA_PAGE_MASK                0x00000FFF
 #define HPFAR_FIPA_PAGE_SHIFT               12
-
+/**
+ * @brief Handles data abort exception.
+ * @param iss ISS is instruction specific syndrome, bits of HSR register 0~24.
+ * @param regs Current ARM registers.
+ * @return If vitual device is successfully emulated, it will be return
```
Comment (Wonseok) [inline hypervisor/trap.h:99]: Return HVMM_STATUS_SUCCESS, If virtual device is successfully emulated.

Comment (Wonseok) [inline hypervisor/trap.h:99]: We don't want to learn details first in this comment.

**Comment by Wonseok** [comment]
Overall looks better than previous. I leave some comments, please take a look.
If you understand my comments enough, then you might want to fix other parts.
Please think more when you write comments.

**Pull request closed without merging**